tools/deviceGen/input/device_limits.json

```diff
@@ -0,0 +1,2347 @@
+            "groupComment": [
+                "VK_KHR_workgroup_memory_explicit_layout",
+                "WorkgroupMemoryExplicitLayoutFeaturesKHR"
+            ]
+        },
+        {
+            "type": "bool",
+            "name": "workgroupMemoryExplicitLayout",
+            "value": false
+        },
+        {
+            "type": "bool",
+            "name": "workgroupMemoryExplicitLayoutScalarBlockLayout",
+            "value": false
+        },
+        {
+            "type": "bool",
+            "name": "workgroupMemoryExplicitLayout8BitAccess",
+            "value": false
+        },
+        {
+            "type": "bool",
+            "name": "workgroupMemoryExplicitLayout16BitAccess",
+            "value": false
+        },
+        {
+            "groupComment": [
+                "[TODO] need new commandbuffer methods, etc",
+                "VK_EXT_color_write_enable",
+                "ColorWriteEnableFeaturesEXT"
+            ]
+        },
+        {
+            "type": "bool",
+            "name": "colorWriteEnable",
+            "value": false
+        },
+        {
+            "groupComment": [
+                "CooperativeMatrixPropertiesKHR",
+                "VK_KHR_cooperative_matrix"
+            ]
+        },
+        {
+            "type": "core::bitflag<asset::IShader::E_SHADER_STAGE>",
+            "name": "cooperativeMatrixSupportedStages",
+            "value": "asset::IShader::ESS_UNKNOWN"
+        }
+    ],
+    "core10": [
+        {
+            "groupComment": [
+                "Always enabled if available, reported as limits",
+                "Core 1.0 Features"
+            ]
+        },
+        {
+            "type": "bool",
+            "name": "logicOp",
+            "value": false,
+            "comment": "mostly just desktops support this"
+        },
+        {
+            "type": "bool",
+            "name": "vertexPipelineStoresAndAtomics",
+            "value": false,
+            "comment": "All iOS GPUs don't support"
+        },
+        {
+            "type": "bool",
+            "name": "fragmentStoresAndAtomics",
+            "value": false,
+            "comment": "ROADMAP 2022 no support on iOS GPUs"
+        },
+        {
+            "type": "bool",
+            "name": "shaderTessellationAndGeometryPointSize",
+            "value": false,
+            "comment": "Candidate for promotion, just need to look into Linux and Android"
+        },
+        {
+            "type": "bool",
+            "name": "shaderStorageImageMultisample",
+            "value": false,
+            "comment": "Apple GPUs and some Intels don't support"
+        },
+        {
+            "type": "bool",
+            "name": "shaderStorageImageReadWithoutFormat",
+            "value": false,
+            "comment": "Intel is a special boy and doesn't support"
+        },
+        {
+            "type": "bool",
+            "name": "shaderStorageImageArrayDynamicIndexing",
+            "value": false,
```

Comment:
if you do https://github.com/Devsh-Graphics-Programming/Nabla/pull/643/files#r1641940730

with the `expose` being an enum, you won't need to declare all these, you'll take them from the features json